File: backend/src/main/java/com/project2/domain/member/service/AuthTokenService.java
package com.project2.domain.member.service;

import com.project2.domain.member.entity.Member;
import com.project2.global.util.Ut;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
@RequiredArgsConstructor
public class AuthTokenService {

    @Value("${custom.jwt.secret-key}")
    private String keyString;

    @Value("${custom.jwt.access-token-expire-seconds}")
    private int accessTokenExpireSeconds;

    @Value("${custom.jwt.refresh-token-expire-seconds}")
    private int refreshTokenExpireSeconds;

    String genAccessToken(Member member) {

        return Ut.Jwt.createToken(
                keyString,
                accessTokenExpireSeconds,
                Map.of("id", member.getId(), "email", member.getEmail())
        );
    }

    public String genRefreshToken(Long id) {
        return Ut.Jwt.createToken(
                keyString,
                refreshTokenExpireSeconds,
                Map.of("id", id)
        );
    }

    Map<String, Object> getPayload(String token) {

        if(!Ut.Jwt.isValidToken(keyString, token)) return null;

        Map<String, Object> payload = Ut.Jwt.getPayload(keyString, token);
        Number idNo = (Number)payload.get("id");
        long id = idNo.longValue();
        String email = (payload.containsKey("email")) ? (String) payload.get("email") : "";

        return Map.of("id", id, "email", email);
    }
}

File: backend/src/main/kotlin/com/project2/domain/member/service/AuthTokenService.kt
package com.project2.domain.member.service

import com.project2.domain.member.entity.Member
import com.project2.global.util.Ut
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class AuthTokenService {

    @Value("\${custom.jwt.secret-key}")
    private lateinit var keyString: String

    @Value("\${custom.jwt.access-token-expire-seconds}")
    private var accessTokenExpireSeconds: Int = 0

    @Value("\${custom.jwt.refresh-token-expire-seconds}")
    private var refreshTokenExpireSeconds: Int = 0

    fun genAccessToken(member: Member): String {
        return Ut.Jwt.createToken(
                keyString,
                accessTokenExpireSeconds,
                mapOf("id" to member.id, "email" to member.email)
        )
    }

    fun genRefreshToken(id: Long): String {
        return Ut.Jwt.createToken(
                keyString,
                refreshTokenExpireSeconds,
                mapOf("id" to id)
        )
    }

    fun getPayload(token: String): Map<String, Any>? {
        if (!Ut.Jwt.isValidToken(keyString, token)) return null

        val payload = Ut.Jwt.getPayload(keyString, token)
        val id = (payload["id"] as Number).toLong()
        val email = payload["email"] as? String ?: ""

        return mapOf("id" to id, "email" to email)
    }
}